modules/features2d/include/opencv2/features2d.hpp

```diff
@@ -271,10 +271,12 @@ class CV_EXPORTS_W SIFT : public Feature2D
 
     @param sigma The sigma of the Gaussian applied to the input image at the octave \#0. If your image
     is captured with a weak camera with soft lenses, you might want to reduce the number.
+
+    @param descriptorType The type of descriptors. Only CV_32F and CV_8U are supported.
     */
     CV_WRAP static Ptr<SIFT> create(int nfeatures = 0, int nOctaveLayers = 3,
         double contrastThreshold = 0.04, double edgeThreshold = 10,
-        double sigma = 1.6);
+        double sigma = 1.6, int descriptorType = CV_32F);
```

Comment:
Could you please add one more `create` function instead. If there are two `create` functions: one with the previous set of parameters and one with extended set of parameters(without default values so all of them are mandatory) it should solve ABI compatibility check issue.

By the way there is no ABI compatibility requirement on master branch, so such a change should pass CI tests there.